### PR TITLE
Breaking Change: Add major and minor to UUID for beacon attribute in beacon monitor

### PR DESCRIPTION
**Pull request by dshokouhi**
## Summary

Fixes: #2908 by adding the `major` and `minor` for the beacon.

Listing as a breaking change since the attribute is changing

## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#n/a

## Any other notes

Tested using the BLE transmitter on other devices to make sure it continues to work and supply valid data.

### CODE REVIEW

File: app/src/main/java/io/homeassistant/companion/android/sensors/BluetoothSensorManager.kt [hunk 1]
```diff
@@ -353,27 +352,22 @@ class BluetoothSensorManager : SensorManager {
 
         val icon = if (monitoringManager.isMonitoring()) "mdi:bluetooth" else "mdi:bluetooth-off"
 
-        var state = if (!BluetoothUtils.isOn(context)) "Bluetooth is turned off" else if (monitoringManager.isMonitoring()) "Monitoring" else "Stopped"
+        val state = if (!BluetoothUtils.isOn(context)) "Bluetooth is turned off" else if (monitoringManager.isMonitoring()) "Monitoring" else "Stopped"
 
-        var attr: Map<String, Any?> = mapOf()
+        val attr: MutableMap<String, Any?> = mutableMapOf()
```
Comment (jpelgrom) [inline app/src/main/java/io/homeassistant/companion/android/sensors/BluetoothSensorManager.kt:357]: Try using type inference if possible.

```suggestion
        val attr = mutableMapOf<String, Any?>()
```

**Review by jpelgrom — COMMENTED** [review]
Should the name format be aligned with core? At first glance, it appears to be [`UUID major.minor`](https://github.com/home-assistant/core/blob/a495df9759ee973620690ee8083d73181547934c/homeassistant/components/ibeacon/coordinator.py#L65) if no specific name is set instead of `UUID-major-minor`.

Also you're joining strings in 4 places, it might be time to create an extension property `name` to avoid the potential for typos in the future.

**Comment by dshokouhi** [comment]
> Should the name format be aligned with core? At first glance, it appears to be [`UUID major.minor`](https://github.com/home-assistant/core/blob/a495df9759ee973620690ee8083d73181547934c/homeassistant/components/ibeacon/coordinator.py#L65) if no specific name is set instead of `UUID-major-minor`.

ah thanks for pointing that out, then maybe we should also update the `ID` attribute for the BLE transmitter to keep it inline?



> Also you're joining strings in 4 places, it might be time to create an extension property `name` to avoid the potential for typos in the future.

good call we should standardize it :)

File: app/src/main/java/io/homeassistant/companion/android/sensors/BluetoothSensorManager.kt (hunk 1 above)
Comment (dshokouhi) [reply app/src/main/java/io/homeassistant/companion/android/sensors/BluetoothSensorManager.kt:357]: will add in my next commit, thank you 🙏

**Comment by jpelgrom** [comment]
> > Should the name format be aligned with core? At first glance, it appears to be [`UUID major.minor`](https://github.com/home-assistant/core/blob/a495df9759ee973620690ee8083d73181547934c/homeassistant/components/ibeacon/coordinator.py#L65) if no specific name is set instead of `UUID-major-minor`.
> 
> ah thanks for pointing that out, then we should also update the `ID` attribute for the BLE transmitter to keep it inline

I hadn't seen that one 🤔 It does explain why you used `-` but I still prefer using the same format as core.

> > Also you're joining strings in 4 places, it might be time to create an extension property `name` to avoid the potential for typos in the future.
> 
> good call we should standardize it :)

Now it is a function available for every class, I was thinking more something like this but that won't work now as it needs to support two classes :)

```kotlin
val IBeacon.name: String
    get() = "$uuid $major.$minor"
```

**Comment by dshokouhi** [comment]
> Now it is a function available for every class, I was thinking more something like this but that won't work now as it needs to support two classes :)

I figure we may also use this in more places so its good that it can used by the app

**Review by jpelgrom — APPROVED** [review]

**Review by JBassett — APPROVED** [review]

**Pull request merged**